elastic/kibana
# [Onboarding] Save installed integrations to the onboarding SO

Closes https://github.com/elastic/kibana/issues/183363

## Summary

Extends the POST/GET /progress endpoints with an `install-integrations` step and it's payload that contains all the installed integrations.

## How to test

1. Run Kibana and ES locally
2. Generate Install API Key in Kibana
3. Go to System logs quick onboarding flow and copy generated ingest API key and onboarding ID
4. Open Network tab in the dev console to track `/progress` endpoint calls
5. Run the auto-detect CLI and go through the flow
```bash
sudo ./x-pack/plugins/observability_solution/observability_onboarding/public/assets/auto_detect.sh --install-key="{{INSTALL_API_KEY}}" --ingest-key="{{INGEST_API_KEY}}" --kibana-url="http://localhost:5601/ftw" --id="{{ONBOARDING_ID}}" --ea-version="8.13.4"
```
 6. Check response from the `/progress` endpoint In the dev console. Make sure there is an `install-integrations` step and it has all of the selected integrations.
